### ISSUE TITLE
"KlaxonException: Need to extract inside" thrown when it shouldn't be

### ISSUE THREAD

**Issue by gps** [cbeust/klaxon#168]
This snippet works as expected:

```kotlin
val jsonString = """
{
    "data": [
        ["a", "b", "c"],
        ["d", "e", "f"]
    ]
}
"""

data class Data(
    val data: List<List<String?>>
)

val klaxon = Klaxon()

val parsed = klaxon.parse<Data>(jsonString)

println(parsed)
```

However, this snippet does not:

```kotlin
val jsonString = """
{
    "data": [
        ["a", "b", null, "c"],
        ["d", "e", null, "f"]
    ]
}
"""

data class Data(
    val data: List<List<String?>>
)

val klaxon = Klaxon()

val parsed = klaxon.parse<Data>(jsonString)

println(parsed)
```

Note the `null`s present in the second snippet.

Stack trace:

```
Caused by: com.beust.klaxon.KlaxonException: Need to extract inside
	at com.beust.klaxon.JsonValue.<init>(JsonValue.kt:83)
	at com.beust.klaxon.DefaultConverter.fromCollection(DefaultConverter.kt:120)
	at com.beust.klaxon.DefaultConverter.fromJson(DefaultConverter.kt:29)
	at com.beust.klaxon.JsonObjectConverter.retrieveKeyValues(JsonObjectConverter.kt:79)
	at com.beust.klaxon.JsonObjectConverter.fromJson(JsonObjectConverter.kt:28)
	at com.beust.klaxon.DefaultConverter.fromJsonObject(DefaultConverter.kt:200)
	at com.beust.klaxon.DefaultConverter.fromJson(DefaultConverter.kt:30)
	at com.beust.klaxon.Klaxon.fromJsonObject(Klaxon.kt:247)
	at KlaxonTest.<init>(KlaxonTest.kts:27)
	... 13 more
```

**Comment by nikolaikopernik** [comment]
Hey, have the same issue. Any ideas how to workaround?  Any plans when it might be solved?

**Comment by cbeust** [comment]
Fixed in 4.0.4.
